faker/providers/barcode/en_CA/__init__.py

```diff
@@ -0,0 +1,15 @@
+from itertools import product
+
+# Canada uses UPC too
+from ..en_US import Provider as BarCodeProvider
+
+
+class Provider(BarCodeProvider):
+    # Source of GS1 country codes: https://gs1.org/standards/id-keys/company-prefix
+    local_prefixes = (
```

Comment:
I am not so sure about storing/specifying prefixes using tuples/lists of tuples/lists. For storing `local_prefixes`, it might be okay to avoid extra conversion steps, but at least for the user API, I think you will agree with this.

```python
# Typing this is quite cumbersome, and there can be too many parentheses
fake.ean13(prefixes=((4, 5), (4, 9)))

# This feels more intuitive and natural
fake.ean13(prefixes=('45', '49'))
```

 What do you think @fcurella?

Reply:
I was wondering what will be the best way to pass it too. I just weren't pretty sure if string splitting and mapping int was the best way to get the prefix. But definitely strings are better on user API.

Also to make the API accept the last example, I can see try doing it in the current implementation  will be a pain.

Reply:
I'm ok with just accepting lists/tuples, as long as the argument is plural (ie: `prefixes`)

Reply:
Say, if I put map(int) around the chosen prefix, will that be better?

```
        if prefixes:
            prefix = self.random_element(prefixes)
            code[:len(prefix)] = map(int, prefix)
```

Reply:
I was also thinking about the last example @malefice showed (`prefixes='45'`), but it might have the potential to make the meaning of the code ambiguous. (`prefixes=((4, 5),)` or `prefixes=((4,), (5,))`)
I mean, I'm sure 99% of the people will understand it as the prior one, but I feel a little uncomfortable.

By adopting the map(int) wrapping above, `prefixes=['45']` or `prefixes={'45'}` will work, so that might work as the more user friendly way of designating a single prefix?

Reply:
Would there be any reason to prevent someone from passing a prefix with a leading zero?

What would happen if a user calls `prefixes=('01', '02')`? Is that a use-case we want to prevent?

Reply:
According to Honeywell(https://support.honeywellaidc.com/s/article/Can-EAN13-symbols-start-with-a-leading-zero), EAN-13 with leading zero is a legitimate code but identical to UPC-A, and there are some barcode readers that treat those codes as UPC-A and omit the first zero when outputting results.

So if someone is generating actual barcode from faker generated codes and is reading with that kind of reader, then it could be a problem. But I don't think it's faker's responsibility to exclude codes with leading zero from valid codes. I think noting the fact that there are those readers and if the user is going to generate the actual barcode, they should know the implications behind it should be enough.

I'll put a little more detail in the docstring and guide to upc code docstring if necessary.

Reply:
@fcurella does this suffice?